### ISSUE TITLE
Ban filter not use full 

### ISSUE THREAD

**Issue by flarecoins** [PHPfox-Official/phpfox-v4-feature-requests#821]
spammer use abc@a.abc.com abc@b.abc.com abc@c.abc.com
this mean .. one website where many subdomain create & then email created ... spammer use this & present phpfox.com not provide help to block domain using wildcard
previous version of phpfox 3+ ... we easily block just *abc.com then full site blocked ... why your team remove this feature not understand.
provide us to help
thank
Anurag

**Comment by harrison05** [comment]
Do you mean Ban Filters > Emails can't add a filter with format *abc.com?

**Comment by flarecoins** [comment]
yes

**Comment by harrison05** [comment]
I think this is an issue, our team will check and update it.

Thanks.

**Comment by harrison05** [comment]
Checking with our development team. Currently, we only support asterisk before @ character for wildcard entries. Example: **@a.abc.com @b.abc.com @c.abc.com**. 
About your request, we will review it later.

**Comment by flarecoins** [comment]
your v3 awesome .. v4 ... your team remove many important setting .. & then your team need money for repair error ... .. its not good sir . & why you review later ? here .. i face lots spammers ... they come & make different different email & i am not ban user with there site name ..  i got 100000+ hit daily .. 2 time my server crash ..its not good ...
or give me instruction .. what i change .. in this file 

--------------------------
<?php
defined('PHPFOX') or exit('NO DICE!');

/**
 * Class Ban_Component_Controller_Admincp_Email
 */
class Ban_Component_Controller_Admincp_Email extends Phpfox_Component
{
	/**
	 * Controller
	 */
	public function process()
	{
		$this->setParam('aBanFilter', array(
				'title' => _p('emails'),
				'type' => 'email',
				'url' => 'admincp.ban.email',
				'form' => _p('email')
			)
		);
		
		return Phpfox_Module::instance()->setController('ban.admincp.default');
	}
	
	/**
	 * Garbage collector. Is executed after this class has completed
	 * its job and the template has also been displayed.
	 */
	public function clean()
	{
		(($sPlugin = Phpfox_Plugin::get('ban.component_controller_admincp_email_clean')) ? eval($sPlugin) : false);
	}
}

[email.txt](https://github.com/PHPfox-Official/phpfox-v4-feature-requests/files/3219250/email.txt)

**Here you see many user just change starting word & same site name.  Its very hard to add all type email in ban filter.** i hope you understand importance of my complain .. suggestion thanks 
![ban1](https://user-images.githubusercontent.com/30006330/58367218-60a72000-7efa-11e9-8bac-3491706af884.png)
![ban2](https://user-images.githubusercontent.com/30006330/58367220-613fb680-7efa-11e9-8153-81dfeb8737f2.png)

**Comment by harrison05** [comment]
Your site gets a hit if they connect to your site even though they don't login/signup, do you try to use ban IP? And you should enable some features to prevent spam email or robot on signup:
- Verify Email At Signup 
- Captcha on Registration (You can use re-captcha)
- Approve Users (This might necessary when your site only allow some emails)

**Comment by flarecoins** [comment]
two year go ... phpfox need just money, no support with important matter, my bad wish to you whole team. you loss your money and time and get result ZERO ... then you feel customers money importance. many time spend for bogus spammers ... you show your full careless work. you support spammers ... this reason never hear customers ... if you serious in your own customers ... you solve this issue in first attempt .. never give 1$ to your team for support ... get from spammers ... where you make loop hole, backdoors ... yes i am angry ... if we face some trouble in sites ... your team open his mouth .. drop some money first ... never recommended phpfox if phpfox continue work this like ...

**Comment by flarecoins** [comment]
yes your last reply fully bogus ...  because you remove important ban feature ... your reply just time pass .

**Comment by flarecoins** [comment]
and IP not belong single person .... then i go to ban direct ip ... its solution ? you developer ? you not understand simple matter ... IP ban not solution, email ban not solution ... only domain name ban is solution .... one thing is important ... some bogus person write this in signup  bogus@bogust.bogus.com ... if you going to check this is domain ? no ... this is only word they use ... how you ban IP ?

**Comment by harrison05** [comment]
Hi @flarecoins 

About Email Filter, with phpFox version 4.8.x, you can use the asterisk (*) and @ for wildcard entries. For example: if you want to ban email:  bogus@bogust.bogus.com, you can input: *@*bogus.com

**Comment by daniol** [comment]
I banned on my site a lot of domain extensions (tld), for example `*@*.online`.
This helped to reduce spam a lot. Normal users don't use such extension for email addresses.
Here is a link with SQL statements:
https://0bin.net/paste/pPagigSU#9QYxiIS6HzkC6C003Uy5ZN0TMX3FOZXGPH0e+bRCWnp
